lib/opencensus/trace/span.rb

```diff
@@ -0,0 +1,198 @@
+      # A unique identifier for a span within a trace, assigned when the span
+      # is created. The ID is an 8-byte value represented as a hexadecimal
+      # string.
+      #
+      # @return [String]
+      #
+      attr_reader :span_id
+
+      ##
+      # The `span_id` of this span's parent span. If this is a root span, then
+      # this field must be empty. The ID is an 8-byte value represented as a
+      # hexadecimal string.
+      #
+      # @return [String]
+      #
+      attr_reader :parent_span_id
+
+      ##
+      # The name of this span.
+      #
+      # @return [OpenCensus::Trace::TruncatableString]
+      #
+      attr_reader :name
+
+      ##
+      # The starting timestamp of this span in UTC, or `nil` if the
+      # starting timestamp has not yet been populated.
+      #
+      # @return [Time, nil]
+      #
+      attr_reader :start_time
+
+      ##
+      # The ending timestamp of this span in UTC, or `nil` if the
+      # ending timestamp has not yet been populated.
+      #
+      # @return [Time, nil]
+      #
+      attr_reader :end_time
+
+      ##
+      # The properties of this span.
+      #
+      # @return [Hash{String => String}]
+      #
+      attr_reader :attributes
+
+      ##
+      # The number of attributes that were discarded. Attributes can be
+      # discarded because their keys are too long or because there are too
+      # many attributes. If this value is 0, then no attributes were dropped.
+      attr_reader :dropped_attributes_count
+
+      ##
+      # A stack trace captured at the start of the span.
+      #
+      # @return [Thread::Backtrace::Location[]]
+      #
+      attr_reader :stack_trace
+
+      ##
+      # The number of stack frames that were dropped because there were too many
+      # stack frames. If this value is 0, then no stack frames were dropped.
+      #
+      # @return [Integer]
+      #
+      attr_reader :dropped_frames_count
+
+      ##
+      # The included time events.
+      #
+      # @return [TimeEvent[]]
+      #
+      attr_reader :time_events
+
+      ##
+      # The number of dropped annotations in all the included time events.
+      # If the value is 0, then no annotations were dropped.
+      #
+      # @return [Integer]
+      #
+      attr_reader :dropped_annotations_count
+
+      ##
+      # The number of dropped message events in all the included time events.
+      # If the value is 0, then no message events were dropped.
+      #
+      # @return [Integer]
+      #
+      attr_reader :dropped_message_events_count
+
+      ##
+      # The included links.
+      #
+      # @return [Link[]]
+      #
+      attr_reader :links
+
+      ##
+      # The number of dropped links after the maximum size was enforced
+      # If the value is 0, then no links were dropped.
+      #
+      # @return [Integer]
+      #
+      attr_reader :dropped_links_count
+
+      ##
+      # An optional final status for this span.
+      #
+      # @return [Status, nil]
```

Comment:
Should we flatten Status? I don't have a strong opinion here, just asking.

Reply:
I think status is optional, but if you have one of the values, then you need both. I think we should keep it.